### PR TITLE
Support party game mode

### LINKED ISSUE #8: Support party game mode
It should be able to create and join a party room by code

### PR DESCRIPTION
**Pull request by xzfc**
Use one of the following scenarios:

* Select `region`, then press `findFfa` or `findParty`.
* Fill `token`, then press `connectParty`.
* Fill `ws` (and optionally fill `token`), then press `directConnect`. May be used for connecting to ogar servers.

Closes #8.

### CODE REVIEW

**Comment by xzfc** [comment]
Screenshot

File: client/js/connector.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+class Connector {
+  constructor() {
+    this.onconnect = () => {};
+    this.onerror = () => {};
+    let that = this;
```
Comment (heroandtn3) [inline client/js/connector.js]: I would like to use `self` instead of `that`

File: client/js/connector.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    this.cb = {
```
Comment (heroandtn3) [inline client/js/connector.js]: What is `cb`? callback?

File: client/js/connector.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }
+  connectParty(token) {
+    findByPartyToken(token, this.cb);
+  }
+  directConnect(ws, token) {
+    this.cb.connect({ ws, token });
+  }
+}
+
+module.exports = Connector;
```
Comment (heroandtn3) [inline client/js/connector.js]: Put this line to the end of the file.

File: client/js/connector.js (hunk 2 above)
Comment (xzfc) [inline client/js/connector.js]: Yes.

File: client/js/connector.js (hunk 3 above)
Comment (xzfc) [inline client/js/connector.js]: Ok. I'd prefer to put exports to very top of the file, but it seems not to
be working.

File: client/js/connector.js (hunk 2 above)
Comment (heroandtn3) [inline client/js/connector.js]: Please use `callback`, saving some characters === making future confusions

**Comment by heroandtn3** [comment]
@xzfc you can merge your PR

**Pull request merged**